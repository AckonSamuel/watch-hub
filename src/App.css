body {
background-color: #070c14;;
}