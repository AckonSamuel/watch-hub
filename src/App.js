import React, { useEffect } from 'react';
import { useSelector, shallowEqual, useDispatch } from 'react-redux';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import Homepage from './components/Homepage';
import Details from './components/details/Details';
import { FetchFilms, FilterFilms, SearchText } from './redux/film';
import './App.css';

const App = () => {
  const dispatch = useDispatch();
  let films = useSelector((state) => state.films.film, shallowEqual);

  useEffect(() => { dispatch(FetchFilms()); }, []);

  const textListener = (title) => {
    dispatch(FilterFilms(title));
    dispatch(SearchText(title));
  };
  const filtered = useSelector((state) => state.films.filters, shallowEqual);
  const searchText = useSelector((state) => state.films.search, shallowEqual);

  if( filtered.length !== 0 )
      { films = filtered;}

console.log(filtered);
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/" element={<Homepage films={films} filtered={filtered} textListener={textListener} searchText={searchText} />} />
        <Route path="/details/:title" element={<Details films={films} />} />
      </Routes>
    </BrowserRouter>
  );
};

export default App;
