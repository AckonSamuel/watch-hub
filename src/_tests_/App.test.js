import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/configureStore';
import  App from '../App';

it('A test for film component', () => {

    const filmTre = renderer.create(
        <Provider store={store}>
            <App />
        </Provider>
    );

    expect(filmTre).toMatchSnapshot();
});