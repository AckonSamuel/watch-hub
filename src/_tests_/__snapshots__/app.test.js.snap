// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A test for film component 1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-light bg-dark"
    style={
      Object {
        "position": "fixed",
        "width": "100%",
        "zIndex": "10000",
      }
    }
  >
    <div
      className="container-fluid"
      style={
        Object {
          "color": "black",
        }
      }
    >
      <a
        className="navbar-brand"
        href="#home"
      >
        Watch-hub
      </a>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <form
          className="d-flex"
        >
          <input
            aria-label="Search"
            className="me-2 form-control"
            onChange={[Function]}
            placeholder="Search Title"
            type="search"
            value=""
          />
        </form>
      </div>
    </div>
  </nav>,
  <div
    className="container"
    style={
      Object {
        "zIndex": "10",
      }
    }
  >
    <div
      className="row"
      style={
        Object {
          "height": "30%",
        }
      }
    >
      <div
        className="card"
        style={
          Object {
            "height": "100%",
            "padding": "0",
            "width": "100%",
          }
        }
      >
        <img
          alt="anime photo"
          className="card-img-top"
          style={
            Object {
              "height": "100%",
              "margin": "0",
              "objectFit": "cover",
              "width": "100%",
            }
          }
        />
      </div>
    </div>
    <div
      className="row"
    >
      <section
        className="container"
      />
    </div>
  </div>,
]
`;
